### DISCUSSION TITLE
Using Newtonsoft.Json in tests: "The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)"

### DISCUSSION THREAD

**Discussion by k7n4n5t3w4rt** [applejag/Newtonsoft.Json-for-Unity#39] [Q&A] [answered]
## Expected behavior

I expected to "using Newtonsoft.Json" to import the package.

## Actual behavior

The "using Newtonsoft" line causes a compilation error:

Assets/Tests/EditMode/DataTest.cs(6,7): error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)

## Steps to reproduce

- New project
- Import `jillejr.newtonsoft.json-for-unity` via UPM
- Add following script to scene:

```csharp
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.TestTools;
using NUnit.Framework;
using Newtonsoft.Json;

public class DataTest
{
  [Test]
  public void JsonIsReturned(){
    var jsonString =  "{\"$base\":14}";
    var expectedInfo = JsonConvert.DeserializeObject<JiraData>(jsonString);
  }
}

```

- Save the file in VSCode and Unity compiles

## Details

Host machine OS running Unity Editor 👉 

Mac OS X 10.14.5
Visual Studio Code 1.41.1
Unity 2019.2.13f1

Unity build target 👉 Andriod/iOS

Newtonsoft.Json-for-Unity package version 👉  12.0.201

I was using Unity version 👉 2019.2.13f1

## Checklist

- [x] Shutdown Unity, deleted the /Library folder, opened project again in Unity, and problem still remains.
- [x] Checked to be using latest version of the package.

![image](https://user-images.githubusercontent.com/358035/72652936-ee310600-39dc-11ea-8349-4423087352b8.png)

Thanks for your package and for your time!

**Comment by k7n4n5t3w4rt** [comment]
Ok. `using Newtonsoft.Json` works for scripts in `/Assets/Scripts/`, just not for the test scripts I've got in `/Assets/Tests/EditMode`. [The solution might be something to do with assembly definition files](https://forum.unity.com/threads/unit-tests-cannot-find-my-namespaces-classes.515742/).

Closing, as this seems to be a Unity thing rather than something specific to this package.

**Comment by applejag** [comment]
Thanks for bringing this up. Yes it seems common to have to specify all references manually for testing assembly definitions.

As said in the linked forum post you can solve it by clicking + on the references and selecting Newtonsoft.Json

**Comment by k7n4n5t3w4rt** [comment, marked as answer]
Oh yes. I wasn't really getting that. In the info panel for the assembly definition file in (for me) the `/Assets/Tests/EditMode` folder. Thanks 🙏🏻

![image](https://user-images.githubusercontent.com/358035/72676318-0fc8e500-3ae4-11ea-8f2e-8e003d41ea38.png)

**Comment by applejag** [comment]
Sometimes (depending on Unity version) disabling the "Override References" checkbox will then automatically reference it, like it does in any other asmdef.
